### DISCUSSION TITLE
Performance Issues with rust ZenEngine when compiled into a C binary

### DISCUSSION THREAD

**Discussion by ishaan812** [gorules/zen#110] [General]
Hey, I'm using Golang and CGo to use zen-engine within my application. Was wondering why the zen-engine slows down significantly when used from a C binary. Why is this an issue?
For example:
For the same decision,
Nodejs performance is :  {14.718079ms map[Level:Rich]}
Whereas, with Go and CGO accesing the rust binary: { performance: '2.8028ms', result: { Level: 'Rich' } 
Couldn't debug why any help would be appreciated!

**Comment by ishaan812** [comment]
Oki nvm, I was running the rust on my WSL and the node on my standard terminal. Not an issue with the C binary/rust enfine. After running both of then on WSL performance is the same. WSL had less access to processing power ig.

**Comment by stefan-gorules** [comment]
Hi @ishaan812 glad it's resolved.

Since you are using Go and CGo wanted to know if you had ideas on how we can make it generally available. We tried in past but since bindings are larger than maximum supported Git size, project cannot be properly uploaded onto Git.

You might find this useful by the way (high level API for Go): https://github.com/gorules/zen-go.
